test/e2e/suite_test.go

```diff
@@ -0,0 +1,142 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/cloud-provider-huaweicloud/pkg/helper"
+	"sigs.k8s.io/cloud-provider-huaweicloud/pkg/util"
+)
+
+const (
+	// RandomStrLength represents the random string length to combine names.
+	RandomStrLength = 5
+)
+
+const (
+	deploymentNamePrefix = "deploy-"
+	serviceNamePrefix    = "service-"
+)
+
+var (
+	// pollInterval defines the interval time for a poll operation.
+	pollInterval time.Duration
+	// pollTimeout defines the time after which the poll operation times out.
+	pollTimeout time.Duration
+)
+
+var (
+	kubeConfig         string
+	restConfig         *rest.Config
+	kubeClient         kubernetes.Interface
+	dynamicClient      dynamic.Interface
+	controlPlaneClient client.Client
+	testNamespace      string
+)
+
+func init() {
+	// usage ginkgo -- --poll-interval=5s --pollTimeout=5m
+	// eg. ginkgo -v --race --trace --fail-fast -p --randomize-all ./test/e2e/ -- --poll-interval=5s --pollTimeout=5m
+	flag.DurationVar(&pollInterval, "poll-interval", 5*time.Second, "poll-interval defines the interval time for a poll operation")
+	flag.DurationVar(&pollTimeout, "poll-timeout", 300*time.Second, "poll-timeout defines the time which the poll operation times out")
+}
+
+func TestE2E(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Skip)
+	ginkgo.RunSpecs(t, "E2E Suite")
+}
+
+var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
+	return nil
+}, func(bytes []byte) {
+	var err error
+
+	kubeConfig = os.Getenv("KUBECONFIG")
+	gomega.Expect(kubeConfig).ShouldNot(gomega.BeEmpty())
+
+	restConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	).ClientConfig()
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	kubeClient, err = kubernetes.NewForConfig(restConfig)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	dynamicClient, err = dynamic.NewForConfig(restConfig)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	controlPlaneClient, err = newForConfigOrDie(restConfig)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	testNamespace = "ccm-test" //fmt.Sprintf("ccmtest-%s", rand.String(RandomStrLength))
```

Comment:
?

Reply:
Sorry, I forgot to delete the test namespace.